crates/wasm-smith/src/core.rs

```diff
@@ -1341,6 +1342,7 @@ pub(crate) fn arbitrary_table_type(u: &mut Unstructured, config: &dyn Config) ->
 }
 
 pub(crate) fn arbitrary_memtype(u: &mut Unstructured, config: &dyn Config) -> Result<MemoryType> {
+    let threads = config.threads_enabled() && u.arbitrary()?;
```

Comment:
I think the `u.arbitrary()` here can be removed since the arbitrary branch for whether or not this memory is shared is generated below in `let shared = ...`

Reply:
For whether or not a feature is enabled we typically leave that exclusively to the `config` rather than factoring in further probability to consider it disabled. To weight the decision of making a shared memory less often I think it might be better to use the `Unstructured::ratio` method perhaps to explicitly indicate that the purpose is to weight the odds not 50/50 but something like 25/75 shared/not-shared